app/Http/Controllers/PrioController.php

```diff
@@ -0,0 +1,473 @@
+                        ])
+                        ->groupBy(['character_items.character_id']);
+                },
+                'wishlistCharacters' => function ($query) use($guild) {
+                    return $query
+                        ->where([
+                            'characters.guild_id' => $guild->id,
+                        ])
+                        ->groupBy(['character_items.character_id']);
+                },
+            ])
+            ->firstOrFail();
+
+        return view('item.prioEdit', [
+            'currentMember' => $currentMember,
+            'guild'         => $guild,
+            'item'          => $item,
+            'maxPrios'      => \App\Http\Controllers\PrioController::MAX_PRIOS,
+            'raid'          => $raid,
+        ]);
+    }
+
+    public function submitMassInput($guildSlug) {
+        $guild         = request()->get('guild');
+        $currentMember = request()->get('currentMember');
+
+        if (!$currentMember->hasPermission('edit.prios')) {
+            request()->session()->flash('status', 'You don\'t have permissions to view that page.');
+            return redirect()->route('member.show', ['guildSlug' => $guild->slug, 'usernameSlug' => $currentMember->slug]);
+        }
+
+        $validationRules =  [
+            'instance_id'           => 'required|exists:instances,id',
+            'items.*.id'            => 'nullable|integer|exists:items,item_id',
+            'items.*.characters.id' => 'nullable|integer|exists:characters,id',
+            'raid_id'               => 'required|exists:raids,id',
+        ];
+
+        $this->validate(request(), $validationRules);
+
+        $guild->load('characters');
+
+        $raid = Raid::where(['guild_id' => $guild->id, 'id' => request()->input('raid_id')])->firstOrFail();
+
+        $instance = Instance::findOrFail(request()->input('instance_id'));
+
+        // Get items for the specified instance WITH any existing prios
+        $itemsWithExistingPrios = Item::select(['items.*'])
+            ->join('item_item_sources', function ($join) {
+                $join->on('item_item_sources.item_id', 'items.item_id');
+            })
+            ->join('item_sources', function ($join) {
+                $join->on('item_sources.id', 'item_item_sources.item_source_id');
+            })
+            ->where([
+                'item_sources.instance_id' => $instance->id,
+            ])
+            ->with([
+                'priodCharacters' => function ($query) use ($raid) {
+                    return $query->where('character_items.raid_id', $raid->id);
+                },
+            ])
+            ->get();
+
+        $modifiedCount = $this->syncPrios($itemsWithExistingPrios, request()->input('items'), $currentMember, $guild->characters, $raid);
+
+        request()->session()->flash('status', 'Successfully updated prios for ' . $modifiedCount . ' items in ' . $raid->name . '.');

```

Comment:
This is pretty neat.

Reply:
Laravel has **so many** features that are damn useful and easy to use like this. https://laravel.com/docs/7.x/session#flash-data

Reply:
This message is presented here: https://github.com/thatsmybis/thatsmybis/blob/master/resources/views/layouts/app.blade.php#L87